controllers/operatorpipeline_controller.go

```diff
@@ -85,10 +135,72 @@ func (r *OperatorPipelineReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		requeueResult = requeueResult || requeue
 	}
 
+	// Adding finalizer to OperatorPipelines CR
+	if !controllerutil.ContainsFinalizer(currentPipeline, operatorPipelineFinalizer) {
+		controllerutil.AddFinalizer(currentPipeline, operatorPipelineFinalizer)
+		if err := r.Update(ctx, currentPipeline); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
 	// Just return the first error reported. It's the most likely issue that needs to be solved.
 	return ctrl.Result{Requeue: requeueResult}, errResult
 }
 
+func (r *OperatorPipelineReconciler) deleteSCCandClusterRole(ctx context.Context, log logr.Logger) error {
+	log.Info("starting deletion of owned SCC and ClusterRole")
+
+	listOption := client.MatchingLabels{
+		fmt.Sprintf("%s/%s", reconcilers.OperatorPipelineLabel, "cluster-resource"): "true",
+	}
+
+	crList := &rbacv1.ClusterRoleList{}
+	if err := r.Client.List(ctx, crList, listOption); err != nil {
+		return err
+	}
+
+	for _, cr := range crList.Items {
+		if err := r.Client.Delete(ctx, &cr); err != nil {
+			return err
+		}
+	}
+
+	sccList := &securityv1.SecurityContextConstraintsList{}
+	if err := r.Client.List(ctx, sccList, listOption); err != nil {
+		return err
+	}
+
+	for _, scc := range sccList.Items {
+		if err := r.Client.Delete(ctx, &scc); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *OperatorPipelineReconciler) deleteClusterRoleBinding(ctx context.Context, log logr.Logger, nameSpace string) error {
+	log.Info("starting deletion of owned ClusterRoleBinding")
+
+	listOption := client.MatchingLabels{
+		fmt.Sprintf("%s/%s", reconcilers.OperatorPipelineLabel, "cluster-resource"): "true",
+		fmt.Sprintf("%s/%s", reconcilers.OperatorPipelineLabel, "metadata.name"):    nameSpace,
```

Comment:
Totally would have just made these constants, and done the fmt.Sprintf-ing at the constant declaration :laughing:

Reply:
`fmt.Sprintf` can't be uses in a constant declaration apparently, but i just made the entire strings constants.